File: api/src/main/java/org/pmiops/workbench/audit/ActionAuditSpringConfiguration.java
package org.pmiops.workbench.audit;

import com.google.cloud.MonitoredResource;
import com.google.cloud.logging.Logging;
import com.google.cloud.logging.LoggingOptions;
import javax.inject.Provider;
import org.pmiops.workbench.config.WorkbenchConfig;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class ActionAuditSpringConfiguration {

  private Provider<WorkbenchConfig> configProvider;

  @Autowired
  ActionAuditSpringConfiguration(Provider<WorkbenchConfig> configProvider) {
    this.configProvider = configProvider;
  }

  @Bean
  public Logging getCloudLogging() {
    return LoggingOptions.getDefaultInstance().getService();
  }

  @Bean
  MonitoredResource getMonitoredResource() {
    final String resourceName = configProvider.get().actionAudit.monitoredResourceName;
    return MonitoredResource.newBuilder(resourceName).build();
  }
}

File: api/src/main/java/org/pmiops/workbench/audit/ActionAuditSpringConfiguration.kt
package org.pmiops.workbench.audit

import com.google.cloud.logging.Logging
import com.google.cloud.logging.LoggingOptions
import org.springframework.beans.factory.config.ConfigurableBeanFactory
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Scope
import java.util.UUID
import javax.inject.Provider

@Configuration
open class ActionAuditSpringConfiguration {

    open val cloudLogging: Logging
        @Bean
        get() = LoggingOptions.getDefaultInstance().service

    open val actionId: String
        @Bean(name = ["ACTION_ID"])
        @Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
        get() {
            return UUID.randomUUID().toString()
        }
}
